homeassistant/components/squeezebox/binary_sensor.py

```diff
@@ -0,0 +1,52 @@
+"""Platform for sensor integration."""
+
+from __future__ import annotations
+
+import logging
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDeviceClass,
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import SqueezeboxConfigEntry
+from .const import STATUS_SENSOR_NEEDSRESTART, STATUS_SENSOR_RESCAN
+from .entity import LMSStatusEntity
+
+SENSORS: tuple[BinarySensorEntityDescription, ...] = (
+    BinarySensorEntityDescription(
+        key=STATUS_SENSOR_RESCAN,
+        device_class=BinarySensorDeviceClass.RUNNING,
+    ),
+    BinarySensorEntityDescription(
+        key=STATUS_SENSOR_NEEDSRESTART,
+        device_class=BinarySensorDeviceClass.UPDATE,
```

Comment:
Both of these sensors are diagnostic so add 
`entity_category=EntityCategory.DIAGNOSTIC,`

Reply:
ok I would agree with entity_category=EntityCategory.DIAGNOSTIC, for needs restart but re-scanning represents a state in the service?

Reply:
Using Unifi Protect as an example I have to hand it has a State = Connected within Diagnostics.
I'd say rescanning is equivalent to that.